packages/console/src/pages/AuditLogDetails/index.tsx

```diff
@@ -0,0 +1,104 @@
+import { LogDTO } from '@logto/schemas';
+import classNames from 'classnames';
+import dayjs from 'dayjs';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import useSWR from 'swr';
+
+import ApplicationName from '@/components/ApplicationName';
+import Card from '@/components/Card';
+import CodeEditor from '@/components/CodeEditor';
+import DetailsSkeleton from '@/components/DetailsSkeleton';
+import LinkButton from '@/components/LinkButton';
+import TabNav, { TabNavItem } from '@/components/TabNav';
+import UserName from '@/components/UserName';
+import { logEventTitle } from '@/consts/logs';
+import { RequestError } from '@/hooks/use-api';
+import Back from '@/icons/Back';
+import * as detailsStyles from '@/scss/details.module.scss';
+
+import EventIcon from './components/EventIcon';
+import * as styles from './index.module.scss';
+
+const AuditLogDetails = () => {
+  const { logId } = useParams();
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { data, error } = useSWR<LogDTO, RequestError>(logId && `/api/logs/${logId}`);
+  const isLoading = !data && !error;
+
+  return (
+    <div className={detailsStyles.container}>
+      <LinkButton
+        to="/audit-logs"
+        icon={<Back />}
+        title="admin_console.log_details.back_to_logs"
+        className={styles.backLink}
+      />
+      {isLoading && <DetailsSkeleton />}
+      {!data && error && <div>{`error occurred: ${error.body?.message ?? error.message}`}</div>}
+      {data && (
+        <>
+          <Card className={styles.header}>
+            <EventIcon isSuccess={data.payload.result === 'Success'} />
+            <div className={styles.content}>
+              <div className={styles.eventName}>{logEventTitle[data.type]}</div>
+              <div className={styles.basicInfo}>
+                <div className={styles.infoItem}>
+                  <div className={styles.label}>{t('log_details.event_type')}</div>
+                  <div>{data.type}</div>
+                </div>
+                <div className={styles.infoItem}>
+                  <div className={styles.label}>{t('log_details.application')}</div>
+                  <div>
+                    {data.payload.applicationId ? (
+                      <ApplicationName isLink applicationId={data.payload.applicationId} />
+                    ) : (
+                      '-'
```

Comment:
nit: It seems that the `<ApplicationName />` component will handle the `-` case.

Reply:
Maybe we could support it and let the `<XXXname>` component handle all the case?